File: app/src/main/java/sk/styk/martin/apkanalyzer/util/networking/BasicAuthInterceptor.java
package sk.styk.martin.apkanalyzer.util.networking;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import java.io.IOException;

import okhttp3.Authenticator;
import okhttp3.Credentials;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;
import okhttp3.Route;

/**
 * Created by Martin Styk on 10.12.2017.
 */
public class BasicAuthInterceptor implements Interceptor {

    private String credentials;
    private static String user = "device";
    private static String password = "******";

    public BasicAuthInterceptor() {
        this.credentials = Credentials.basic(user, password);
    }

    @Override
    public Response intercept(@NonNull Chain chain) throws IOException {
        Request request = chain.request();
        Request authenticatedRequest = request.newBuilder()
                .header("Authorization", credentials).build();
        return chain.proceed(authenticatedRequest);
    }

}

File: app/src/main/java/sk/styk/martin/apkanalyzer/util/networking/BasicAuthInterceptor.kt
package sk.styk.martin.apkanalyzer.util.networking

import okhttp3.Credentials
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

/**
 * Interceptor for authentication to server
 * This interceptor adds auth header to every request
 *
 * @author Martin Styk
 * @version 10.12.2017.
 */
class BasicAuthInterceptor : Interceptor {

    private val credentials: String = Credentials.basic(USER, PASSWORD)

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val request = chain.request()
        val authenticatedRequest = request.newBuilder()
                .header("Authorization", credentials).build()
        return chain.proceed(authenticatedRequest)
    }

    companion object {
        private const val USER = "device"
        private const val PASSWORD = "******"
    }

}